MantidQt/API/test/NonOrthogonalTest.h

```diff
@@ -224,124 +79,215 @@ class NonOrthogonalTest : public CxxTest::TestSuite {
+          Mantid::DataObjects::MDEvent<4>, 4>(1, 0.0, 1.0, frame, 1, wsName);
+    }
+    if (!forgetUB) {
+      // add UB Matrix
+      Mantid::API::IAlgorithm_sptr alg =
+          Mantid::API::AlgorithmManager::Instance().create("SetUB");
+      alg->initialize();
+      alg->setRethrows(true);
+      alg->setProperty("Workspace", wsName);
+      alg->setProperty("a", 3.643 * scale);
+      alg->setProperty("b", 3.643);
+      alg->setProperty("c", 5.781);
+      alg->setProperty("alpha", 90.0);
+      alg->setProperty("beta", 90.0);
+      alg->setProperty("gamma", 120.0);
+      std::vector<double> uVec;
+      uVec.push_back(1 * scale);
+      uVec.push_back(1);
+      uVec.push_back(0);
+      std::vector<double> vVec;
+      vVec.push_back(0);
+      vVec.push_back(0);
+      vVec.push_back(1);
+      alg->setProperty("u", uVec);
+      alg->setProperty("v", vVec);
+      alg->execute();
+    }
+
+    // Create the coordinate transformation information
+    std::vector<Mantid::coord_t> affMatVals{1, 0, 0, 0, 0, 0, 0, 1, 0,
+                                            0, 0, 0, 0, 1, 0, 0, 1, 0,
+                                            0, 0, 0, 0, 0, 0, 1};
+
+    Mantid::DataObjects::CoordTransformAffine affMat(4, 4);
+    affMat.setMatrix(Mantid::Kernel::Matrix<Mantid::coord_t>(affMatVals));
+    if (!forgetAffmat) {
+      ws->setTransformToOriginal(affMat.clone(), 0);
+    }
+    // Create the transform (W) matrix
+    // Need it as a vector
+    std::vector<double> wMat;
+    Mantid::Kernel::DblMatrix temp(3, 3, true);
+    wMat = temp.getVector();
+    if (!forgetWmat) {
+      auto p = new Mantid::Kernel::PropertyWithValue<std::vector<double>>(
+          "W_MATRIX", wMat);
+      ws->getExperimentInfo(0)->mutableRun().addProperty(p, true);
+    }
+    return ws;
+  }
+
+  Mantid::Kernel::DblMatrix getExampleSkewMatrix() {
+    Mantid::Kernel::DblMatrix skewMatrix(3, 3, true);
+    skewMatrix[0][0] = 1;
+    skewMatrix[0][1] = 0;
+    skewMatrix[0][2] = -0.57735;
+    skewMatrix[1][0] = 0;
+    skewMatrix[1][1] = 1;
+    skewMatrix[1][2] = 0;
+    skewMatrix[2][0] = 0;
+    skewMatrix[2][1] = 0;
+    skewMatrix[2][2] = 1.1547;
+    return skewMatrix;
+  }
+
+  template <typename T> bool skewWithinTolerance(T coord, T target) {
+    if ((coord >= target - 0.000005) && (coord <= target + 0.000005)) {
+      return true;
+    } else
+      return false;
+  }
+
+  std::array<Mantid::coord_t, 9> getExampleCoordTArray(bool nonSkewed) {
+    std::array<Mantid::coord_t, 9> coordinates;
+    if (nonSkewed) {
+      coordinates = {{1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0}};
+    } else {
+      coordinates = {{1.0, 0.0, static_cast<Mantid::coord_t>(-0.57735), 0.0,
+                      1.0, 0.0, 0.0, 0.0,
+                      static_cast<Mantid::coord_t>(1.1547)}};
+    }
+    return coordinates;
+  }
+
+  enum DimensionIndex { DimX = 0, DimY = 1, SliceDim = 2 };
```

Comment:
`DimensionIndex` itself is not but `DimX` etc are used throughout the test. This is a plain old `enum` so that you can avoid `static_casts` everywhere as we just want to use it as an `unsigned`. This means no qualification with the `enum` name is necessary.